### PR TITLE
DOC: scan area detector with motor(s)

### LINKED ISSUE #308: HOWTO: Scan area detector v. motor(s), view data afterwards
For commissioning activities, users want to make engineering scans with area detectors for various purposes and view the data from those scans right away.

Make a HOWTO document with these goals:

- [ ] use EPICS motor(s)
- [ ] use EPICS area detector
  - [ ] For each image
    - [ ] Write image to HDF5 file.
    - [ ] Measure total counts in image.
    - [ ] Measure maximum counts in image.
    - [ ] Note the image's "unique ID".
- [ ] Step scan motor and collect $n$ image frame(s) at each step.
  - [ ] count
  - [ ] scan with one motor
  - [ ] scan with two motors
  - [ ] grid_scan
- [ ] Collect data in a SPEC data file.

### PR DESCRIPTION
**Pull request by prjemian**
- close #308 

@sureshnaps, @qzhang - Can you examine this notebook?  It can guide you during commissioning activities.

Start here: https://bcda-aps.github.io/bluesky_training/howto/_ad_v_motors.html#The-(1-D)-bp.scan()-plan-with-one-motor

### CODE REVIEW

**Comment by rodolakis** [comment]
In general, I find some of the markdown comment difficult to find in the middle of the code cell output, like here for example:

image


maybe adding bullet point or a `<br>` before the comment would help.

**Comment by rodolakis** [comment]
Not sure what that bug was in the last few suggestions. It was displaying the correct sentence before pressing the green button, but then showed the code for an image as the original...

**Review by rodolakis — CHANGES_REQUESTED** [review]
Minor formatting and rephrasing.

File: docs/source/howto/_ad_v_motors.ipynb [hunk 1]
```diff
@@ -0,0 +1,3485 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Configure the area detector for:\n",
+    "\n",
+    "- Collect $n$ image frame(s) for each press of the *Acquire* button.\n",
+    "- Save image(s) to HDF5\n",
+    "- Record total counts in the image.\n",
+    "- Record maximum counts in the image.\n",
+    "- Record the unique ID of the image."
```
[the remark this answers is not in the document]
Comment (prjemian) [reply docs/source/howto/_ad_v_motors.ipynb:218]: Not the right word here.  "Configure the detector **_for_** Collect $n$ image frame(s) ..."

File: docs/source/howto/_ad_v_motors.ipynb [hunk 2]
```diff
@@ -0,0 +1,3485 @@
...
+    "\n",
+    "These steps are optional."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Show the data structure collected by the `RE` during a routine data acquisition\n",
+    "run with the area detector object `simdet`."
```
[the remark this answers is not in the document]
Comment (prjemian) [reply docs/source/howto/_ad_v_motors.ipynb:265]: Such as:

> Show the data from area detector `simdet`.  This is the data collected by the `RE` during a run.

File: docs/source/howto/_ad_v_motors.ipynb [hunk 3]
```diff
@@ -0,0 +1,3485 @@
...
+    "    det.image.stage_sigs[\"blocking_callbacks\"] = \"No\"\n",
+    "\n",
+    "ad_setup(simdet)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Diagnostics (optional)\n",
```
[the remark this answers is not in the document]
Comment (prjemian) [reply docs/source/howto/_ad_v_motors.ipynb:255]: I'll take this point and see how to apply generally.

File: docs/source/howto/_ad_v_motors.ipynb [hunk 4]
```diff
@@ -0,0 +1,3485 @@
...
...mKfc3Wobam9rXJ5FWhCgB+NqOztJqxCnfPlHHy3ooIxYP7VpWxLss7L6/VGhAjSY3hAMA8w7ko+VYk55sdxn5P3pC4u6v9gNvvzU7E9yk4SbuKyJeq6kQtmnZV11JUSw9tlYoyMUwnLGzReR4x/mdgxpmpqc3TTFWvZi562N5/CGqwOx2PtPpstVrhD//wD+3/Cmicnp7iUY96FJ773Ofib//tv42/8Tf+Bp785CfjFa94BX76p38ar3rVqwBcBzRu3LiB09NTfO3Xfu01QONdNVxKj+CZew8ei8UC0+kUT84+B0V/1El+BwObALUg0htXH9BazOiNqEN5REC3utOCwYh+GpApDxZXlsYlUm8NncB8v2+ZRTqBKoFauT0ADOZXSb1+rvXIU5LCrye5ZtPOb0UmUu3fx+3WigxdUWY3z2xF1643VjC2l/OO37TdXfM30ngDzUbSoSo0C1gVywKV2epxP4xPAWC+JdnNs87D6aBXrR5DWihafEbZZZ+prYK6jFvRKr5Tei5Ugt/eun1Nip+2Oz64xeTOHcD3abG0B61yheLVnGRimYQRPAugGPmwFFWbPqz1Qeo00BUgyVmuH42a0OLBSTitFjDWllEekRYv+9rQHABEjUpRK96+gLtxysKmaYkyyUMOKbEY0kLvoN2n1gwoC2Cx7D57OKCf0eUV92O7A/o9FkKTMYu6PDN0DLvqmv+RtTKF82YPKeFvpfXGbAvcZEzuy4FU+/Bv/cnM0L54Nbdzoi7JADoER1qTdu2qR4+aY0p0it374pOl/DQtRmxhJCKLQ9m78VX0elBivhaFeWbXeHN+cc1fK9b0U0v7vfHkIPesGRFKsa2fDRxkzh16EsnCTucAF8I1LiT2tRX9ds/tKhYQohQ9XNRohIa13uUztMhyisTrfAaxK9DFhPogifzfOFXzBecWIa6H0dDmr7hYXfMmCuOxoWw6f7k8s4Wo3pvG/RKvIwCdolja7PzwZB5Gep7reoP/sv1xzOfzazydh3Poc+l3f+dujB9C9tlyGfGBj/vzd3hbX/WqV10DNHQooAEA3/d934cXvvCFeOMb34jHPvaxeP7zn4+nPe1p9trdbofnPve5+LEf+zEDNL7ne77HlN7vqnEsit7BoRffJxSfC1+zjWWtiAMzMXOMBcyAzZVELWylKoWLISOA3aCaHK+TSRhJa+tg0j
```
[the remark this answers is not in the document]
Comment (prjemian) [reply docs/source/howto/_ad_v_motors.ipynb:2075]: I don't understand this review comment.

[the remark this answers is not in the document]
Comment (prjemian) [reply docs/source/howto/_ad_v_motors.ipynb:2587]: I don't understand this review comment.

[the remark this answers is not in the document]
Comment (prjemian) [reply docs/source/howto/_ad_v_motors.ipynb:2598]: I don't understand this review comment.

[the remark this answers is not in the document]
Comment (prjemian) [reply docs/source/howto/_ad_v_motors.ipynb:3145]: I don't understand this review comment.

[the remark this answers is not in the document]
Comment (prjemian) [reply docs/source/howto/_ad_v_motors.ipynb]: I don't understand this review comment.

**Comment by prjemian** [comment]
@rodolakis I believe this is ready for review again.

File: docs/source/howto/_ad_v_motors.ipynb (hunk 4 above)
[the remark this answers is not in the document]
Comment (prjemian) [reply docs/source/howto/_ad_v_motors.ipynb:2097]: > Not sure what that bug was in the last few suggestions. It was displaying the correct sentence before pressing the green button, but then showed the code for an image as the original...

Probably these next comments.

**Comment by rodolakis** [comment]
OK so this is what I was trying to suggest:

image

**Comment by rodolakis** [comment]
image


I would write instead:

    "Both motors, `m1` and `m2`, move at the same time. Below is a plot showing the readback values of each motor during a two-motor `scan` versus the data point number:\n",

**Comment by rodolakis** [comment]
> The file name, as the catalog will find its image contents, is reconstructed:

I am not sure I understand this sentence. I was proposing:

"The catalog uses the file name to locate and retrieve its image contents. This file name is reconstructed as follows:"

**Comment by rodolakis** [comment]
> "The second motor of the `grid_scan` (`m2`) is snaked.  It's direction changes with each subsequent increment of the
> first motor (`m1`).  Here's a plot of readback value of each motor from a `grid_scan` plotted *v*. data point number:\n",

I was proposing:

    "During the `grid_scan`, the second motor (`m2`) moves in a snaking pattern: its direction changes with each subsequent increment of the first motor (`m1`). Below is a plot showing the readback values of each motor during a `grid_scan` versus the data point number:\n",

**Comment by rodolakis** [comment]
image



- I was proposing instead:


    All along, quietly in the background, the data has been saved both to a catalog
    and to a SPEC data file. The total counts, max counts, and unique ID from all
    scans can be plotted from the SPEC file using PyMCA or NeXpy. The image content
    is not stored in the SPEC file, so images cannot be viewed from it. However,
    references to the images are save as comments.  See the comment lines that
    match a search for `resource {'spec': 'AD_HDF5'}`, such as this one:

**Comment by rodolakis** [comment]
@prjemian  I rewrote my broken comments. I still cannot quote anything as a github "suggestions".

**Comment by prjemian** [comment]
@rodolakis : I believe I have responded to all your comments.  Thanks for the thorough review.

File: docs/source/howto/_ad_v_motors.ipynb [hunk 5]
```diff
@@ -0,0 +1,3521 @@
...
+    "## Diagnostics (optional)\n",
+    "\n",
+    "These steps are optional."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Show the data from area detector `simdet`.\n",
```
[the remark this answers is not in the document]
Comment (rodolakis) [reply docs/source/howto/_ad_v_motors.ipynb:264]: @prjemian  this is the only comment left that has not been resolved. Minor though, but would be nice to add it.

**Comment by rodolakis** [comment]
All good. Everything has been resolved as far as I am concerned except for one minor formatting suggestion (I tagged you on it).
So minor that I'm fine with merging with or without.

**Review by rodolakis — APPROVED** [review]

**Comment by prjemian** [comment]
@rodolakis Thanks!

**Pull request merged**